src/global/signal_handler.cc

```diff
@@ -313,13 +313,33 @@ struct SignalHandler : public Thread {
 	    r = read(handlers[signum]->pipefd[0], &v, 1);
 	    if (r == 1) {
 	      siginfo_t * siginfo = &handlers[signum]->info_t;
-	      string task_name = get_name_by_pid(siginfo->si_pid);
-	      derr << "received  signal: " << sig_str(signum)
-		   << " from " << " PID: " << siginfo->si_pid
-		   << " task name: " << task_name
-		   << " UID: " << siginfo->si_uid
-		   << dendl;
-	      handlers[signum]->handler(signum);
+              ostringstream message;
+              message << "received  signal: " << sig_str(signum);
+              switch (siginfo->si_code) {
+                case SI_USER:
+                  message << " from " << get_name_by_pid(siginfo->si_pid);
```

Comment:
i'd suggest update `get_name_by_pid()`, and return "kernel" if `pid` is 0. please note, Ceph is portable and runs on  FreeBSD also.

Reply:
That's exactly the aim of the first patch of this PR.
I updated it to return "Kernel" instead of "Linux Kernel".